From 396cbe3848850062fe160ba5de28e44fd7898adc Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 28 Apr 2015 18:28:53 +0800
Subject: [PATCH 2/2] tweak backend.py/gui.py/package_gui.py/task_gui.py to
 support package-selection

gui.py/package_gui.py:
  Rename 'GroupSelectionWindow' to 'PackageSelectionWindow';

backend.py/task_gui.py/package_gui.py
  While package-selection used, installing the selected package to
  target, and do not break previous installation flow.

Upstream-Status: Inappropriate [WRLinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/backend.py        |  3 +++
 pyanaconda/gui.py            |  2 +-
 pyanaconda/iw/package_gui.py | 22 +++++++++++++++++-----
 pyanaconda/iw/task_gui.py    |  1 +
 4 files changed, 22 insertions(+), 6 deletions(-)

diff --git a/pyanaconda/backend.py b/pyanaconda/backend.py
index 2adb728..0972223 100644
--- a/pyanaconda/backend.py
+++ b/pyanaconda/backend.py
@@ -154,6 +154,9 @@ class AnacondaBackend:
     def deselectGroup(self, group, *args):
         log.warning("deselectGroup not implemented for backend!")
 
+    def resetGroup(self):
+        log.warning("resetGroup not implemented for backend!")
+
     def packageExists(self, pkg):
         log.warning("packageExists not implemented for backend!")
         return True
diff --git a/pyanaconda/gui.py b/pyanaconda/gui.py
index 1d8a05c..e81626d 100644
--- a/pyanaconda/gui.py
+++ b/pyanaconda/gui.py
@@ -74,7 +74,7 @@ stepToClass = {
     "timezone" : ("timezone_gui", "TimezoneWindow"),
     "accounts" : ("account_gui", "AccountWindow"),
     "tasksel": ("task_gui", "TaskWindow"),
-    "group-selection": ("package_gui", "GroupSelectionWindow"),
+    "group-selection": ("package_gui", "PackageSelectionWindow"),
     "install" : ("progress_gui", "InstallProgressWindow"),
     "complete" : ("congrats_gui", "CongratulationWindow"),
 }
diff --git a/pyanaconda/iw/package_gui.py b/pyanaconda/iw/package_gui.py
index 1e8492c..7a66cd9 100644
--- a/pyanaconda/iw/package_gui.py
+++ b/pyanaconda/iw/package_gui.py
@@ -20,7 +20,7 @@
 #
 
 # FIXME: group selection isn't currently backend independent
-from GroupSelector import GroupSelector
+from PackageSelector import PackageSelector
 
 from pyanaconda import gui
 from iw_gui import *
@@ -29,11 +29,23 @@ from pyanaconda.constants import *
 import gettext
 _ = lambda x: gettext.ldgettext("anaconda", x)
 
-class GroupSelectionWindow (InstallWindow):
+class PackageSelectionWindow (InstallWindow):
     def getScreen(self, anaconda):
         self.backend = anaconda.backend
         self.intf = anaconda.intf
-        self.grpsel = GroupSelector(self.backend.ayum, gui.findGladeFile,
+        self.pkgsel = PackageSelector(self.backend.asmart, gui.findGladeFile,
                                     gui.addFrame)
-        self.grpsel.doRefresh()
-        return self.grpsel.vbox
+        self.pkgsel.doRefresh()
+        return self.pkgsel.vbox
+
+    def getNext(self):
+        self.backend.resetGroup()
+
+        # The extra packages make sure the target could boot
+        for pkg in ["packagegroup-wr-base", "packagegroup-wr-boot"]:
+            if self.pkgsel.packages.get_available_package(pkg):
+                self.backend.selectPackage(pkg)
+
+        for pkg in self.pkgsel.getSelectedPackages():
+            self.backend.selectPackage(pkg)
+
diff --git a/pyanaconda/iw/task_gui.py b/pyanaconda/iw/task_gui.py
index 608b43e..c2b5a98 100644
--- a/pyanaconda/iw/task_gui.py
+++ b/pyanaconda/iw/task_gui.py
@@ -529,6 +529,7 @@ class TaskWindow(InstallWindow):
 
         if self.xml.get_widget("customRadio").get_active():
             self.dispatch.request_steps("group-selection")
+            return
         else:
             self.dispatch.skip_steps("group-selection")
 
-- 
1.9.1

